Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+JavaBindingsPrinting.swift

```diff
@@ -113,20 +131,29 @@ extension JNISwift2JavaGenerator {
   }
 
   private func printProtocol(_ printer: inout CodePrinter, _ decl: ImportedNominalType) {
-    let extends = ["JNISwiftInstance"]
-    printer.printBraceBlock("public interface \(decl.swiftNominal.name) extends \(extends.joined(separator: ", "))") { printer in
+    var extends = [String]()
+
+    // If we cannot generate Swift wrappers
+    // that allows the user to implement the wrapped interface in Java
+    // then we require only JExtracted types can conform to this.
+    if !self.interfaceProtocolWrappers.keys.contains(decl) {
+      extends.append("JNISwiftInstance")
+    }
+    let extendsString = extends.isEmpty ? "" : " extends \(extends.joined(separator: ", "))"
+
+    printer.printBraceBlock("public interface \(decl.swiftNominal.name)\(extendsString)") { printer in
       for initializer in decl.initializers {
-        printFunctionDowncallMethods(&printer, initializer, signaturesOnly: true)
+        printFunctionDowncallMethods(&printer, initializer, signaturesOnly: true, ignoresArenas: true)
```

Comment:
```suggestion
        printFunctionDowncallMethods(&printer, initializer, signaturesOnly: true, ignoreArenas: true)
```

`ignores` sounds weird there